sip/DockComponentsFactory.sip

```diff
@@ -1,27 +1,31 @@
 namespace ads
 {
-    
+
 class CDockComponentsFactory
 {
-
-    %TypeHeaderCode
-    #include <DockComponentsFactory.h>
-    %End
+%TypeHeaderCode
+#include <DockComponentsFactory.h>
+%End
 
 public:
-	virtual ~CDockComponentsFactory();
-	virtual CDockWidgetTab* createDockWidgetTab(CDockWidget* DockWidget /Transfer/ ) const;
-    virtual CAutoHideTab* createDockWidgetSideTab(CDockWidget* DockWidget /Transfer/) const;
-	virtual CDockAreaTabBar* createDockAreaTabBar(CDockAreaWidget* DockArea /Transfer/ ) const;
-	virtual CDockAreaTitleBar* createDockAreaTitleBar(CDockAreaWidget* DockArea /Transfer/ ) const;
+  virtual ~CDockComponentsFactory();
+  virtual CDockWidgetTab* createDockWidgetTab(CDockWidget* DockWidget /Transfer/) const;
+  virtual CAutoHideTab* createDockWidgetSideTab(CDockWidget* DockWidget /Transfer/) const;
+  virtual CDockAreaTabBar* createDockAreaTabBar(CDockAreaWidget* DockArea /Transfer/) const;
+  virtual CDockAreaTitleBar* createDockAreaTitleBar(CDockAreaWidget* DockArea /Transfer/) const;
+  static const CDockComponentsFactory* factory();
+%MethodCode
+    QSharedPointer<ads::CDockComponentsFactory> *grab;
+
+    Py_BEGIN_ALLOW_THREADS
+    // This will leak but there seems to be no way to detach the object.
+    grab = new QSharedPointer<ads::CDockComponentsFactory>(ads::CDockComponentsFactory::factory());
+    Py_END_ALLOW_THREADS
 
-    static CDockComponentsFactory* factory();
-    %MethodCode
-        sipRes = ::ads::CDockComponentsFactory::factory().data();
-    %End
-	static void setFactory(CDockComponentsFactory* Factory /KeepReference/);
-	static void resetDefaultFactory();
-    
+    sipRes = grab->data();
```

Comment:
@char101, this was one method I had naively updated just to get it to build ... I assume what you've done here is preferable right?

Reply:
It was copied from the PyQt6 source. I am not sure about the significance of moving the shared pointer either. I don't have any problem when compiling it though (I was using clang-cl on Windows).

sip\QtQuick\qquickitem.sip
```
public:
    void resetAntialiasing();
    QQuickItemGrabResult *grabToImage(const QSize &targetSize = QSize()) /Factory/;
%MethodCode
        QSharedPointer<QQuickItemGrabResult> *grab;
        
        Py_BEGIN_ALLOW_THREADS
        // This will leak but there seems to be no way to detach the object.
        grab = new QSharedPointer<QQuickItemGrabResult>(sipCpp->grabToImage(*a0));
        Py_END_ALLOW_THREADS
        
        sipRes = grab->data();
%End

    bool isAncestorOf(const QQuickItem *child) const;
```

Reply:
Also in DockManager.sip

```
CDockComponentsFactory* componentsFactory() const;
%MethodCode
    sipres = sipCpp->componentsFactory().data();
%End
```